api/novelist.py

```diff
@@ -433,6 +460,73 @@ def get_recommendations(self, metadata, recommendations_info):
                 metadata.recommendations += self._extract_isbns(book_info)
         return metadata
 
+    def get_isbns_from_query(self, library):
+        collectionList = []
+        for c in library.collections:
+            collectionList.append(c.id)
+
+        LEFT_OUTER_JOIN = True
+        i1 = aliased(Identifier)
+        i2 = aliased(Identifier)
+        e = aliased(Equivalency)
+        lp = aliased(LicensePool)
+
+        isbnQuery = select(
+            [i1.identifier, i1.type, i2.identifier]
+        ).select_from(
+            join(lp, i1, i1.id==lp.identifier_id)
+            .join(e, i1.id==e.input_id, LEFT_OUTER_JOIN)
+            .join(i2, e.output_id==i2.id, LEFT_OUTER_JOIN)
+        ).where(
+            and_(
+                lp.collection_id.in_(collectionList),
+                or_(i1.type=="ISBN", i2.type=="ISBN")
+            )
+        ).alias('lp_isbns')
+
+        result = self._db.execute(isbnQuery)
+
+        isbns = []
+        for res in result:
+            if (res[1] == "ISBN"):
+                isbns.append(res[0])
+            else:
+                isbns.append(res[2])
+
+        return isbns
+
+    def put_isbns_novelist(self, library):
+        isbns = self.get_isbns_from_query(library)
+
+        content = None
+        if isbns:
+            response = self.put(
+                self.COLLECTION_DATA_API,
+                { "AuthorizedIdentifier": self.authorized_identifier },
```

Comment:
Given that the credentials we got from NoveList in the most recent email include both a profile and a password, It's confusing to me why only the 'password' is necessary here.

Reply:
I'm using the `authorized identifier` as a header, but I'm using the `profile` and `password` in the data object request that gets sent in line 507.